blueocean-core-js/src/js/services/DefaultSSEHandler.js

```diff
@@ -66,12 +66,18 @@ export class DefaultSSEHandler {
 
         for (const key of this.branchPagerKeys(event)) {
             const pager = this.pagerService.getPager({ key });
-            this.activityService.fetchActivity(runSelf, { overrideQueuedState }).then(d => {
-                if (pager && !pager.has(runSelf)) {
-                    pager.insert(runSelf);
-                }
-                this.pipelineService.updateLatestRun(d);
-            });
+            // If we don't have a pager, then the app in the browser has never visited that
+            // Job, which means there's no need to fetch the data associated with it because
+            // there's nothing in the app that needs to be kept up-to-date. The data will
+            // be fetched when the user visits that job/pipeline.
+            if (pager) {
+                this.activityService.fetchActivity(runSelf, { overrideQueuedState }).then(d => {
+                    if (!pager.has(runSelf)) {
+                        pager.insert(runSelf);
+                    }
+                    this.pipelineService.updateLatestRun(d);
```

Comment:
I'm just pointing out that I think this is what could be causing the ATH failure you mentioned. I like the change, but there probably needs to be some more logic to determine what to update (this is exactly why I've mentioned not being a fan of this manual client cache we have). I think it's a valid improvement, but as you said, probably not critical.